### ISSUE TITLE
Video texture provider, texture 2D, ECT needs file extension in the URL in order to work

### ISSUE THREAD

**Issue by epicEaston197** [Neos-Metaverse/NeosPublic#3918]
### Is your tweak request related to a problem? Please describe.

The issue is that if I don't have dot MP4 or some other video file extension at the end of my video URL the video will refuse to load

### Describe what would you like tweaked

I would like neos to not require a file extension at the end of URL's because for example if I have a MP4 without dot MP4 in a video player based on context it should know it's a video since it's in a video texture provider

### Describe alternatives you've considered

Basically detecting if the URL has dot MP4 at the end if it doesn't insert dot MP4 so the video works

### Additional context

I am making a Mastodon viewer and sometimes the URL doesn't have dot MP4 resulting in the video just not loading

**Comment by JackTheFoxOtter** [comment]
There is more video formats / containers than just mp4 out there. You can't assume every resource leads to a .mp4 file just because it's used in the context of the video player. That said, when accessing web resources, usually the type of the resource is specified in the MIME-Type of the response header (for example 'video/mp4' or 'video/webm'). If Neos doesn't already, it could use that to determine what kind of video it is, and choose the corresponding converter if no file extension is present in the URL. This naturally would only work if the correct MIME-Type is specified in the external resource though.

Edit:
Actually, for all web resources it should definitely prioritize the MIME type over a possible file extension in the URL, as that's what web browsers use to determine how a resource should be handled / displayed.
See possible subtypes here:
https://www.iana.org/assignments/media-types/media-types.xhtml#video

**Comment by Firepal** [comment]
MIME types were an afterthought? in a web-centric application which can load files from remote servers? truly amazing
